hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FairSchedulerConfiguration.java

```diff
@@ -287,27 +289,69 @@ public float getReservableNodes() {
    * 
    * @throws AllocationConfigurationException
    */
-  public static Resource parseResourceConfigValue(String val)
+  public static ResourceConfiguration parseResourceConfigValue(String val)
       throws AllocationConfigurationException {
+    ResourceConfiguration resourceConfiguration;
     try {
       val = StringUtils.toLowerCase(val);
-      int memory = findResource(val, "mb");
-      int vcores = findResource(val, "vcores");
-      return BuilderUtils.newResource(memory, vcores);
+      if (val.contains("%")) {
+        resourceConfiguration = new ResourceConfiguration(
+            getResourcePercentage(val));
+      } else {
+        int memory = findResource(val, "mb");
+        int vcores = findResource(val, "vcores");
+        resourceConfiguration = new ResourceConfiguration(
+            BuilderUtils.newResource(memory, vcores));
+      }
     } catch (AllocationConfigurationException ex) {
       throw ex;
     } catch (Exception ex) {
       throw new AllocationConfigurationException(
           "Error reading resource config", ex);
     }
+    return resourceConfiguration;
+  }
+
+  private static HashMap<ResourceType, Double> getResourcePercentage(
+      String val) throws AllocationConfigurationException {
+    HashMap<ResourceType, Double> resourcePercentage = new HashMap<>();
+    String[] strings = val.split(",");
+    if (strings.length == 1) {
+      double percentage = findPercentage(strings[0], "");
+      for (ResourceType resourceType: ResourceType.values()) {
+        resourcePercentage.put(resourceType, percentage/100);
+      }
+    } else {
+      double memPercentage = findPercentage(val, "memory");
+      double vcorePercentage = findPercentage(val, "vcore");
```

Comment:
vcore is a unit of cpu. If we are using memory instead of mb, we should use cpu instead of vcore.